# Import Library

In [ ]:
import numpy as np
import pandas as pd
import pickle
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


# Download Stopwords Indonesia

In [ ]:
stopwords_ind = stopwords.words('indonesian')
len(stopwords_ind)

758

# Import Dataset

In [ ]:
data_tourism = pd.read_csv('tourism_with_id.csv')
data_tourism

Place_Id                                 Place_Name  \
0           1                           Monumen Nasional   
1           2                                   Kota Tua   
2           3                              Dunia Fantasi   
3           4          Taman Mini Indonesia Indah (TMII)   
4           5                   Atlantis Water Adventure   
..        ...                                        ...   
432       433                        Museum Mpu Tantular   
433       434                              Taman Bungkul   
434       435           Taman Air Mancur Menari Kenjeran   
435       436               Taman Flora Bratang Surabaya   
436       437  Gereja Perawan Maria Tak Berdosa Surabaya   

                                           Description       Category  \
0    Monumen Nasional atau yang populer disingkat d...         Budaya   
1    Kota tua di Jakarta, yang juga bernama Kota Tu...         Budaya   
2    Dunia Fantasi atau disebut juga Dufan adalah t...  Taman Hiburan   
3    Taman Mini Indonesia Indah merupakan suatu kaw...  Taman Hiburan   
4    Atlantis Water Adventure atau dikenal dengan A...  Taman Hiburan   
..                                                 ...            ...   
432  Museum Negeri Mpu Tantular adalah sebuah museu...         Budaya   
433  Taman Bungkul adalah taman wisata kota yang te...  Taman Hiburan   
434  Air mancur menari atau dancing fountain juga a...  Taman Hiburan   
435  Taman Flora adalah salah satu taman kota di Su...  Taman Hiburan   
436  Gereja Katolik Kelahiran Santa Perawan Maria m...  Tempat Ibadah   

         City   Price  Rating  Time_Minutes  \
0     Jakarta   20000     4.6          15.0   
1     Jakarta       0     4.6          90.0   
2     Jakarta  270000     4.6         360.0   
3     Jakarta   10000     4.5           NaN   
4     Jakarta   94000     4.5          60.0   
..        ...     ...     ...           ...   
432  Surabaya    2000     4.4          45.0   
433  Surabaya       0     4.6           NaN   
434  Surabaya       0     4.4          45.0   
435  Surabaya       0     4.6           NaN   
436  Surabaya   10000     4.8           NaN   

                                          Coordinate       Lat        Long  \
0            {'lat': -6.1753924, 'lng': 106.8271528} -6.175392  106.827153   
1    {'lat': -6.137644799999999, 'lng': 106.8171245} -6.137645  106.817125   
2    {'lat': -6.125312399999999, 'lng': 106.8335377} -6.125312  106.833538   
3    {'lat': -6.302445899999999, 'lng': 106.8951559} -6.302446  106.895156   
4               {'lat': -6.12419, 'lng': 106.839134} -6.124190  106.839134   
..                                               ...       ...         ...   
432          {'lat': -7.4338593, 'lng': 112.7199058} -7.433859  112.719906   
433  {'lat': -7.291346799999999, 'lng': 112.7398218} -7.291347  112.739822   
434          {'lat': -7.2752955, 'lng': 112.7549381} -7.275296  112.754938   
435  {'lat': -7.294330299999999, 'lng': 112.7617534} -7.294330  112.761753   
436          {'lat': -7.2420758, 'lng': 112.7368158} -7.242076  112.736816   

     Unnamed: 11  Unnamed: 12  
0            NaN            1  
1            NaN            2  
2            NaN            3  
3            NaN            4  
4            NaN            5  
..           ...          ...  
432          NaN          433  
433          NaN          434  
434          NaN          435  
435          NaN          436  
436          NaN          437  

[437 rows x 13 columns]

In [ ]:
data_tourism.duplicated().value_counts()

False    437
dtype: int64

# Data Preparation

In [ ]:
data_tourism = data_tourism[data_tourism.City == 'Bandung']
data_tourism.reset_index(inplace = True, drop = True)

In [ ]:
data_tourism.head()

Place_Id                       Place_Name  \
0       211           GunungTangkuban perahu   
1       212                      Jalan Braga   
2       213                      Gedung Sate   
3       214             Trans Studio Bandung   
4       215  Taman Hutan Raya Ir. H. Djuanda   

                                         Description       Category     City  \
0  Gunung Tangkuban Parahu adalah salah satu gunu...     Cagar Alam  Bandung   
1  Jalan Braga adalah nama sebuah jalan utama di ...         Budaya  Bandung   
2  Gedung Sate, dengan ciri khasnya berupa orname...         Budaya  Bandung   
3  Trans Studio Bandung adalah kawasan wisata ter...  Taman Hiburan  Bandung   
4  Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     Cagar Alam  Bandung   

    Price  Rating  Time_Minutes  \
0   30000     4.5           NaN   
1       0     4.7           NaN   
2    5000     4.6           NaN   
3  280000     4.5         240.0   
4   15000     4.5          90.0   

                                        Coordinate       Lat        Long  \
0  {'lat': -6.759637700000001, 'lng': 107.6097807} -6.759638  107.609781   
1          {'lat': -6.9150534, 'lng': 107.6089842} -6.915053  107.608984   
2            {'lat': -6.9024812, 'lng': 107.61881} -6.902481  107.618810   
3          {'lat': -6.9250943, 'lng': 107.6364944} -6.925094  107.636494   
4          {'lat': -6.8565791, 'lng': 107.6323734} -6.856579  107.632373   

   Unnamed: 11  Unnamed: 12  
0          NaN          211  
1          NaN          212  
2          NaN          213  
3          NaN          214  
4          NaN          215

In [ ]:
data_tourism.shape

(124, 13)

In [ ]:
data_tourism['tags'] = data_tourism['Category'] + ' ' + data_tourism['Description']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
data_tourism.head()

Place_Id                       Place_Name  \
0       211           GunungTangkuban perahu   
1       212                      Jalan Braga   
2       213                      Gedung Sate   
3       214             Trans Studio Bandung   
4       215  Taman Hutan Raya Ir. H. Djuanda   

                                         Description       Category     City  \
0  Gunung Tangkuban Parahu adalah salah satu gunu...     Cagar Alam  Bandung   
1  Jalan Braga adalah nama sebuah jalan utama di ...         Budaya  Bandung   
2  Gedung Sate, dengan ciri khasnya berupa orname...         Budaya  Bandung   
3  Trans Studio Bandung adalah kawasan wisata ter...  Taman Hiburan  Bandung   
4  Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     Cagar Alam  Bandung   

    Price  Rating  Time_Minutes  \
0   30000     4.5           NaN   
1       0     4.7           NaN   
2    5000     4.6           NaN   
3  280000     4.5         240.0   
4   15000     4.5          90.0   

                                        Coordinate       Lat        Long  \
0  {'lat': -6.759637700000001, 'lng': 107.6097807} -6.759638  107.609781   
1          {'lat': -6.9150534, 'lng': 107.6089842} -6.915053  107.608984   
2            {'lat': -6.9024812, 'lng': 107.61881} -6.902481  107.618810   
3          {'lat': -6.9250943, 'lng': 107.6364944} -6.925094  107.636494   
4          {'lat': -6.8565791, 'lng': 107.6323734} -6.856579  107.632373   

   Unnamed: 11  Unnamed: 12                                               tags  
0          NaN          211  Cagar Alam Gunung Tangkuban Parahu adalah sala...  
1          NaN          212  Budaya Jalan Braga adalah nama sebuah jalan ut...  
2          NaN          213  Budaya Gedung Sate, dengan ciri khasnya berupa...  
3          NaN          214  Taman Hiburan Trans Studio Bandung adalah kawa...  
4          NaN          215  Cagar Alam Taman Hutan Raya Ir. H. Djuanda (Ta...

In [ ]:
# data_tourism['index'] = range(0, (len(data_tourism)-1) + 1)
# data_tourism.head()

In [ ]:
# first_column = data_tourism.pop('index')
# data_tourism.insert(0, 'index', first_column)
# data_tourism
# data_tourism_new = data_tourism[['Place_Id','Place_Name','tags']]

In [ ]:
data_tourism_new = data_tourism[['Place_Id','Place_Name','Category','Description','Rating','tags']]

In [ ]:
data_tourism_new.head()

Place_Id                       Place_Name       Category  \
0       211           GunungTangkuban perahu     Cagar Alam   
1       212                      Jalan Braga         Budaya   
2       213                      Gedung Sate         Budaya   
3       214             Trans Studio Bandung  Taman Hiburan   
4       215  Taman Hutan Raya Ir. H. Djuanda     Cagar Alam   

                                         Description  Rating  \
0  Gunung Tangkuban Parahu adalah salah satu gunu...     4.5   
1  Jalan Braga adalah nama sebuah jalan utama di ...     4.7   
2  Gedung Sate, dengan ciri khasnya berupa orname...     4.6   
3  Trans Studio Bandung adalah kawasan wisata ter...     4.5   
4  Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     4.5   

                                                tags  
0  Cagar Alam Gunung Tangkuban Parahu adalah sala...  
1  Budaya Jalan Braga adalah nama sebuah jalan ut...  
2  Budaya Gedung Sate, dengan ciri khasnya berupa...  
3  Taman Hiburan Trans Studio Bandung adalah kawa...  
4  Cagar Alam Taman Hutan Raya Ir. H. Djuanda (Ta...

In [ ]:
data_tourism_new

Place_Id                       Place_Name       Category  \
0         211           GunungTangkuban perahu     Cagar Alam   
1         212                      Jalan Braga         Budaya   
2         213                      Gedung Sate         Budaya   
3         214             Trans Studio Bandung  Taman Hiburan   
4         215  Taman Hutan Raya Ir. H. Djuanda     Cagar Alam   
..        ...                              ...            ...   
119       330                Bandros City Tour         Budaya   
120       331          Kyotoku Floating Market         Budaya   
121       332                   Rainbow Garden     Cagar Alam   
122       333                        Kota Mini  Taman Hiburan   
123       334         Chingu Cafe Little Seoul  Taman Hiburan   

                                           Description  Rating  \
0    Gunung Tangkuban Parahu adalah salah satu gunu...     4.5   
1    Jalan Braga adalah nama sebuah jalan utama di ...     4.7   
2    Gedung Sate, dengan ciri khasnya berupa orname...     4.6   
3    Trans Studio Bandung adalah kawasan wisata ter...     4.5   
4    Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     4.5   
..                                                 ...     ...   
119  Bandros atau Bus Wisata Bandung adalah bus wis...     4.4   
120  Kyotoku Floating Market Bandung, sebuah wisata...     4.5   
121  Rainbow Garden Harapan Indah salah satu taman ...     4.6   
122  Destinasi yang sangat menarik bernuansa eropa ...     4.4   
123  Selain populer karena memiliki pemandangan yan...     4.5   

                                                  tags  
0    Cagar Alam Gunung Tangkuban Parahu adalah sala...  
1    Budaya Jalan Braga adalah nama sebuah jalan ut...  
2    Budaya Gedung Sate, dengan ciri khasnya berupa...  
3    Taman Hiburan Trans Studio Bandung adalah kawa...  
4    Cagar Alam Taman Hutan Raya Ir. H. Djuanda (Ta...  
..                                                 ...  
119  Budaya Bandros atau Bus Wisata Bandung adalah ...  
120  Budaya Kyotoku Floating Market Bandung, sebuah...  
121  Cagar Alam Rainbow Garden Harapan Indah salah ...  
122  Taman Hiburan Destinasi yang sangat menarik be...  
123  Taman Hiburan Selain populer karena memiliki p...  

[124 rows x 6 columns]

In [ ]:
data_tourism_new.isnull().sum()

Place_Id       0
Place_Name     0
Category       0
Description    0
Rating         0
tags           0
dtype: int64

In [ ]:
pickle.dump(data_tourism_new,open('data_tourism_new.pkl','wb'))

# Modelling

In [ ]:
#cv = CountVectorizer()

In [ ]:
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(stop_words=stopwords_ind, lowercase=True)

In [ ]:
vector = cv.fit_transform(data_tourism_new['tags']).toarray()

/usr/local/lib/python3.7/dist-packages/sklearn/feature_extraction/text.py:401: UserWarning: Your stop_words may be inconsistent with your preprocessing. Tokenizing the stop words generated tokens ['baiknya', 'berkali', 'kali', 'kurangnya', 'mata', 'olah', 'sekurang', 'setidak', 'tama', 'tidaknya'] not in stop_words.
  % sorted(inconsistent)


In [ ]:
#vector = cv.fit_transform(data_tourism_new['Category']).toarray()

In [ ]:
vector

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 1, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]])

In [ ]:
vector.shape

(124, 2625)

In [ ]:
from sklearn.metrics.pairwise import cosine_similarity

In [ ]:
similarity = cosine_similarity(vector)

In [ ]:
pickle.dump(similarity,open('similarity.pkl','wb'))

In [ ]:
from google.colab.data_table import DataTable
DataTable.max_columns = 123

In [ ]:
df_similarity = pd.DataFrame(similarity)
df_similarity

0         1         2         3         4         5         6    \
0    0.000000  0.075023  0.115875  0.069586  0.182989  0.024098  0.215061   
1    0.075023  0.000000  0.133660  0.214043  0.145928  0.111187  0.289414   
2    0.115875  0.133660  0.000000  0.110197  0.119694  0.048532  0.143680   
3    0.069586  0.214043  0.110197  0.000000  0.225586  0.160422  0.357917   
4    0.182989  0.145928  0.119694  0.225586  0.000000  0.110487  0.324941   
..        ...       ...       ...       ...       ...       ...       ...   
119  0.123274  0.255950  0.132728  0.322394  0.220940  0.164426  0.387231   
120  0.033697  0.134746  0.074128  0.134593  0.061799  0.169796  0.167132   
121  0.041739  0.064194  0.034486  0.105851  0.057991  0.185577  0.138013   
122  0.000000  0.102233  0.111216  0.063215  0.011082  0.177325  0.021979   
123  0.046179  0.213066  0.075117  0.241540  0.125110  0.188206  0.209953   

          7         8         9    ...       114       115       116  \
0    0.188178  0.108442  0.044026  ...  0.067118  0.080648  0.029691   
1    0.041345  0.092656  0.090282  ...  0.095286  0.196388  0.076108   
2    0.082252  0.098932  0.042060  ...  0.059188  0.108280  0.052131   
3    0.051131  0.148964  0.062804  ...  0.068740  0.300395  0.141183   
4    0.168073  0.331462  0.055051  ...  0.047343  0.201688  0.074253   
..        ...       ...       ...  ...       ...       ...       ...   
119  0.086618  0.127887  0.077881  ...  0.078284  0.305709  0.082537   
120  0.055711  0.069916  0.091238  ...  0.059917  0.161561  0.098450   
121  0.103510  0.082479  0.213469  ...  0.000000  0.230022  0.186307   
122  0.021979  0.019703  0.107988  ...  0.008443  0.087917  0.137563   
123  0.038173  0.102658  0.093775  ...  0.058651  0.209953  0.168646   

          117       118       119       120       121       122       123  
0    0.238806  0.089719  0.123274  0.033697  0.041739  0.000000  0.046179  
1    0.133556  0.068993  0.255950  0.134746  0.064194  0.102233  0.213066  
2    0.136211  0.027799  0.132728  0.074128  0.034486  0.111216  0.075117  
3    0.206460  0.051194  0.322394  0.134593  0.105851  0.063215  0.241540  
4    0.316706  0.052354  0.220940  0.061799  0.057991  0.011082  0.125110  
..        ...       ...       ...       ...       ...       ...       ...  
119  0.255112  0.040812  0.000000  0.153281  0.063288  0.030237  0.192553  
120  0.053988  0.096684  0.153281  0.000000  0.115332  0.132244  0.162689  
121  0.037152  0.092123  0.063288  0.115332  0.000000  0.091003  0.158052  
122  0.000000  0.044013  0.030237  0.132244  0.091003  0.000000  0.132146  
123  0.107897  0.025480  0.192553  0.162689  0.158052  0.132146  1.000000  

[124 rows x 124 columns]

In [ ]:
df_mostsimilar = df_similarity

In [ ]:
for i in range (0,123):
  df_mostsimilar[i] = df_mostsimilar[i].replace(to_replace = df_mostsimilar[i].max(), value = 0)

In [ ]:
for i in range (0,123):
  print(df_mostsimilar[i].max())

0.5281207860194961
0.4197675776868771
0.4222960557762796
0.4725359454076892
0.9656724005383486
0.37565274200835697
0.4474990813863584
0.5570860145311556
0.3391231227194379
0.2707832187790668
0.42733800419231627
0.3835109258067629
0.6012276399448193
0.43388133406288953
0.2300655576751468
0.4967951322546442
0.41216786985544673
0.37363235887853663
0.3312147284572834
0.3892801930051676
0.37565274200835697
0.4018486467184276
0.5184369937141206
0.3663967458131496
0.2964683161313088
0.8163118322929529
0.5533167449931865
0.5281207860194961
0.20919054414739632
0.6332827240238033
0.30879033974997094
0.44348306662723397
0.49263482989964497
0.32474295832352745
0.3187276291558383
0.46459032891518787
0.7519154820750275
0.25900594339123895
0.24051807590126403
0.33490586430504626
0.5771942074840609
0.372450346494751
0.26737691438783046
0.4780089717611234
0.5120915564991891
0.3919930312969426
0.36590203268178384
0.5120915564991891
0.3301478222570351
0.42315179501837635
0.2737189818695862
0.416651989083

Nilai cosine similarity tertinggi adalah 0.8163118322929529 antara Taman Sejarah Bandung dan Taman Balai Kota Bandung

In [ ]:
data_tourism_new.loc[[74, 25]]

Place_Id                Place_Name       Category  \
74       285     Taman Sejarah Bandung         Budaya   
25       236  Taman Balai Kota Bandung  Taman Hiburan   

                                          Description  Rating  \
74  Taman Sejarah Bandung adalah taman yang terlet...     4.6   
25  Taman Balai Kota Bandung merupakan sebuah tama...     4.6   

                                                 tags  
74  Budaya Taman Sejarah Bandung adalah taman yang...  
25  Taman Hiburan Taman Balai Kota Bandung merupak...

In [ ]:
def recommend1(place):
  model_df = similarity
  model_df = pd.DataFrame(model_df, index=data_tourism_new['Place_Name'], columns=data_tourism_new['Place_Name'])
  count = 5
  items=data_tourism_new[['Place_Name','Description','Rating']]
  index = model_df.loc[:,place].to_numpy().argpartition(range(-1, -count, -1))
  closest = model_df.columns[index[-1:-(count+2):-1]]
  closest = closest.drop(place, errors='ignore')
  new_rec = pd.DataFrame(closest).merge(items).head(count)

  list_of_movie = new_rec['Place_Name'].tolist()
  list_of_rating = new_rec['Rating'].tolist()
  list_of_description = new_rec['Description'].tolist()

  return list_of_movie,list_of_rating,list_of_description

In [ ]:
recommend1('Farm House Susu Lembang')

(['The Great Asia Africa',
  'Peta Park',
  'Rabbit Town',
  'Happyfarm Ciwidey',
  'Dago Dreampark'],
 [4.4, 4.3, 4.0, 4.2, 4.2],
 ['Kota Bandung tampaknya tidak pernah kehabisan ide kreatif dalam mengembangkan destinasi wisata. Berbagai tujuan berlibur tersedia di sini, mulai dari wisata kuliner, sejarah, hingga edukasi. Kemasannya pun dengan cara yang unik dan menarik, seperti yang ada di The Great Asia Africa. Kawasan wisata teranyar di Lembang ini menawarkan pengalaman menjelajahi beberapa negara di dua benua, selaras dengan namanya.',
  'Bandung seperti tidak henti menawarkan tempat wisata menarik untuk dikunjungi. Terutama untuk para anak muda. Beberapa waktu lalu Peta Park Bandung dibuka. Tempat wisata yang menawarkan konsep taman-taman yang bersih dan indah. Sangat cocok sebagai tempat bersantai dan menghindari hiruk-pikuk kota. Taman rekreasi yang ada di pusat kota Bandung ini memiliki konsep taman piknik. Karena hal tersebut, saat berada di sini, pengunjung tidak hanya bisa 

In [ ]:
data_tourism_new

Place_Id                       Place_Name       Category  \
0         211           GunungTangkuban perahu     Cagar Alam   
1         212                      Jalan Braga         Budaya   
2         213                      Gedung Sate         Budaya   
3         214             Trans Studio Bandung  Taman Hiburan   
4         215  Taman Hutan Raya Ir. H. Djuanda     Cagar Alam   
..        ...                              ...            ...   
119       330                Bandros City Tour         Budaya   
120       331          Kyotoku Floating Market         Budaya   
121       332                   Rainbow Garden     Cagar Alam   
122       333                        Kota Mini  Taman Hiburan   
123       334         Chingu Cafe Little Seoul  Taman Hiburan   

                                           Description  Rating  \
0    Gunung Tangkuban Parahu adalah salah satu gunu...     4.5   
1    Jalan Braga adalah nama sebuah jalan utama di ...     4.7   
2    Gedung Sate, dengan ciri khasnya berupa orname...     4.6   
3    Trans Studio Bandung adalah kawasan wisata ter...     4.5   
4    Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     4.5   
..                                                 ...     ...   
119  Bandros atau Bus Wisata Bandung adalah bus wis...     4.4   
120  Kyotoku Floating Market Bandung, sebuah wisata...     4.5   
121  Rainbow Garden Harapan Indah salah satu taman ...     4.6   
122  Destinasi yang sangat menarik bernuansa eropa ...     4.4   
123  Selain populer karena memiliki pemandangan yan...     4.5   

                                                  tags  
0    Cagar Alam Gunung Tangkuban Parahu adalah sala...  
1    Budaya Jalan Braga adalah nama sebuah jalan ut...  
2    Budaya Gedung Sate, dengan ciri khasnya berupa...  
3    Taman Hiburan Trans Studio Bandung adalah kawa...  
4    Cagar Alam Taman Hutan Raya Ir. H. Djuanda (Ta...  
..                                                 ...  
119  Budaya Bandros atau Bus Wisata Bandung adalah ...  
120  Budaya Kyotoku Floating Market Bandung, sebuah...  
121  Cagar Alam Rainbow Garden Harapan Indah salah ...  
122  Taman Hiburan Destinasi yang sangat menarik be...  
123  Taman Hiburan Selain populer karena memiliki p...  

[124 rows x 6 columns]

In [ ]:
data_tourism_new

Place_Id                       Place_Name       Category  \
0         211           GunungTangkuban perahu     Cagar Alam   
1         212                      Jalan Braga         Budaya   
2         213                      Gedung Sate         Budaya   
3         214             Trans Studio Bandung  Taman Hiburan   
4         215  Taman Hutan Raya Ir. H. Djuanda     Cagar Alam   
..        ...                              ...            ...   
119       330                Bandros City Tour         Budaya   
120       331          Kyotoku Floating Market         Budaya   
121       332                   Rainbow Garden     Cagar Alam   
122       333                        Kota Mini  Taman Hiburan   
123       334         Chingu Cafe Little Seoul  Taman Hiburan   

                                           Description  Rating  \
0    Gunung Tangkuban Parahu adalah salah satu gunu...     4.5   
1    Jalan Braga adalah nama sebuah jalan utama di ...     4.7   
2    Gedung Sate, dengan ciri khasnya berupa orname...     4.6   
3    Trans Studio Bandung adalah kawasan wisata ter...     4.5   
4    Taman Hutan Raya Ir. H. Djuanda (Tahura Djuand...     4.5   
..                                                 ...     ...   
119  Bandros atau Bus Wisata Bandung adalah bus wis...     4.4   
120  Kyotoku Floating Market Bandung, sebuah wisata...     4.5   
121  Rainbow Garden Harapan Indah salah satu taman ...     4.6   
122  Destinasi yang sangat menarik bernuansa eropa ...     4.4   
123  Selain populer karena memiliki pemandangan yan...     4.5   

                                                  tags  
0    Cagar Alam Gunung Tangkuban Parahu adalah sala...  
1    Budaya Jalan Braga adalah nama sebuah jalan ut...  
2    Budaya Gedung Sate, dengan ciri khasnya berupa...  
3    Taman Hiburan Trans Studio Bandung adalah kawa...  
4    Cagar Alam Taman Hutan Raya Ir. H. Djuanda (Ta...  
..                                                 ...  
119  Budaya Bandros atau Bus Wisata Bandung adalah ...  
120  Budaya Kyotoku Floating Market Bandung, sebuah...  
121  Cagar Alam Rainbow Garden Harapan Indah salah ...  
122  Taman Hiburan Destinasi yang sangat menarik be...  
123  Taman Hiburan Selain populer karena memiliki p...  

[124 rows x 6 columns]